README.md

```diff
@@ -55,13 +57,165 @@ And in your modules, apply the `android-apt` plugin and add the remixer dependen
 apply plugin: 'android-apt'
 
 dependencies {
-    compile 'com.github.material-foundation:material-remixer-android:develop-SNAPSHOT'
+    compile 'com.github.material-foundation.material-remixer-android:remixer_core:develop-SNAPSHOT'
+    compile 'com.github.material-foundation.material-remixer-android:remixer_ui:develop-SNAPSHOT'
+    compile 'com.github.material-foundation.material-remixer-android:remixer_storage:develop-SNAPSHOT'
     provided 'com.github.material-foundation.material-remixer-android:remixer_annotation:develop-SNAPSHOT'
 }
 ```
 
 Notice the dependency on `remixer_annotation` is a `provided` clause instead of `compile`, this is on purpose as this is not a regular dependency but a compiler plugin.
 
+### Global remixer set up
+If you have not subclassed the application class it is recommended you do it since this is a one-time global initialization.
+
+In your application class you need to add RemixerActivityLifecycleCallbacks as an ActivityLifecycleCallbacks instance, so Remixer knows to remove old variables and triggers when activities are destroyed, to avoid leaks.
+```java
+class MyApplication extends android.app.Application {
+  @Override
+  public void onCreate() {
+    super.onCreate();
+    registerActivityLifecycleCallbacks(RemixerActivityLifecycleCallbacks.getInstance());
+  }
+}
+```
+
+### How to use it in an activity
+
+__Only in the activities where you're using remixer__
+
+You need to add a few lines at the end of your `Activity.onCreate()`
+
+1. `RemixerBinder.bind(this);` creates, initializes and sets up all the Variables and trigger you define in this activity.
+2. `RemixerFragment remixerFragment = RemixerFragment.newInstance();` creates the fragment that will be shown when remixer is invoked, then you need at least one of the following:
+  - A variation of `remixerFragment.attachToGesture(this, Direction.UP, 3);`, this example ties showing the Remixer Fragment on a 3-finger swipe up.
+  - `remixerFragment.attachToButton(this, someButtonObject);`, this makes the OnClickListener for a button open the Remixer fragment.
+
+Your `Activity.onCreate` may look like this:
```

Comment:
same advice for this section. Although you may want to keep *some* explanation, most of this can be replaced with "Learn more about the APIs defined in the library by reading our [technical documentation](https://jitpack.io/com/github/material-foundation/material-remixer-android/0.5.0/javadoc/)" (you can add the link once you have javadoc generated)